src/Microsoft.Extensions.Logging.EventSource/EventSourceLoggerProvider.cs

```diff
@@ -12,226 +13,94 @@ namespace Microsoft.Extensions.Logging.EventSource
-                    return LogLevel.Information;
-                }
+                return LogLevel.Trace;
             }
-            else
-            {
-                if (_eventSource.IsEnabled(EventLevel.Warning, allMessageKeywords))
-                {
-                    return LogLevel.Warning;
-                }
-                else
-                {
-                    if (_eventSource.IsEnabled(EventLevel.Error, allMessageKeywords))
-                    {
-                        return LogLevel.Error;
-                    }
-                    else
-                    {
-                        return LogLevel.Critical;
-                    }
-                }
-            }
-        }
 
-        /// <summary>
-        /// Given a set of specifications  Pat1:Level1;Pat1;Level2 ... Where
-        /// Pat is a string pattern (a logger Name with a optional trailing wildcard * char)
-        /// and Level is a number 0 (Trace) through 5 (Critical).
-        ///
-        /// The :Level can be omitted (thus Pat1;Pat2 ...) in which case the level is 1 (Debug).
-        ///
-        /// A completely emtry sting act like * (all loggers set to Debug level).
-        ///
-        /// The first speciciation that 'loggers' Name matches is used.
-        /// </summary>
-        private LogLevel ParseLevelSpecs(string filterSpec, LogLevel defaultLevel, string loggerName)
-        {
-            if (filterSpec == null)
-            {
-                return LoggingEventSource.LoggingDisabled;      // Null means disable.
-            }
-            if (filterSpec == string.Empty)
+            var rules = _rules;
+            var level = LogLevel.None;
+            foreach (var rule in rules)
             {
-                return defaultLevel;
-            }
-
-            var level = LoggingEventSource.LoggingDisabled;   // If the logger does not match something, it is off.
+                Debug.Assert(rule.LogLevel.HasValue);
+                Debug.Assert(rule.ProviderName == GetType().FullName);
 
-            // See if logger.Name  matches a _filterSpec pattern.
-            var namePos = 0;
-            var specPos = 0;
-            for (;;)
-            {
-                if (namePos < loggerName.Length)
+                if (rule.CategoryName == null)
                 {
-                    if (filterSpec.Length <= specPos)
-                    {
-                        break;
-                    }
-                    var specChar = filterSpec[specPos++];
-                    var nameChar = loggerName[namePos++];
-                    if (specChar == nameChar)
-                    {
-                        continue;
-                    }
-
-                    // We allow wildcards at the end.
-                    if (specChar == '*' && ParseLevel(defaultLevel, filterSpec, specPos, ref level))
-                    {
-                        return level;
-                    }
+                    level = rule.LogLevel.Value;
```

Comment:
Should this be checking if the `rule.LogLevel` is more restrictive than the current value? Otherwise it will just apply the last rule only.

Reply:
It would take first matching rule the same way EventLogger worked before the change.

Reply:
Only rules coming from `ParseFilterSpec` are passed in this method and only one of them wouldn't have category.